src/AndroidDebugLauncher/Launcher.cs

```diff
@@ -556,37 +556,63 @@ private void VerifySdkVersion()
-            // The lib directory symlinks to somewhere else on the file system, for example /data/data/<appname>/lib/ -> /data/app-lib/<appname>-1/lib
-            // On 64 bit, this symlink is broken (I think this is a bug in 64 bit android).
-            // We must attempt to find the gdbserver path manually then. 
+            // On Android, the files of an application are extracted to the system owned directory /data/app/<appname>-<int>/lib
+            // For 32-bit shared libraries, Android will create a symlink from the app owned directory /data/data/<appname>/lib
+            // to maintain backwards compatibility with previous API versions.
 
-            // Android issue: https://code.google.com/p/android/issues/detail?id=186010&thanks=186010&ts=1441922626
+            // There is no such symlink for 64-bit libraries, since Android only supports 64-bit since API 21, at which point
+            // the app owned directory was no longer used for libs.
 
-            //check the correct location for x86/arm/arm64
+            // check the legacy location for 32-bit libs
             string gdbServerPath = workingDirectory + "/lib/gdbserver";
 
             string lsCommand = string.Format(CultureInfo.InvariantCulture, "ls {0}", gdbServerPath);
             string output = ExecCommand(lsCommand);
-            if (string.Compare(output, gdbServerPath, StringComparison.OrdinalIgnoreCase) == 0)
-            {
-                //gdbserver is symlinked correctly
-                return gdbServerPath;
-            }
-            else if (_launchOptions.TargetArchitecture == TargetArchitecture.X64)
-            {
-                //start looking other places, only do this on 64 bit
-                //TODO: This needs some additional testsing
-                lsCommand = string.Format(CultureInfo.InvariantCulture, "ls /data/app/{0}*/lib/x86_64/gdbserver", _launchOptions.Package);
-                output = ExecCommand(lsCommand);
-                return output;
-            }
-            else
-            {
-                throw new LauncherException(Telemetry.LaunchFailureCode.NoGdbServer, string.Format(CultureInfo.CurrentCulture, LauncherResources.Error_NoGdbServer, gdbServerPath));
+            if (string.Compare(output, gdbServerPath, StringComparison.Ordinal) != 0)
+            {
+                // <app data>/lib is not symlinked to system's app data directory (/data/app/<app package>/lib). 
+                // We copy gdbserver ourselves because of 2 reasons:
+                // 1. the app user doesn't have permissions on /data/app/<app package>-<suffix>/lib (it's owned by system)
+                // 2. we can't deterministically figure out the location where gdbserver is in /data/app because of the suffix,
+                //    which is used by Android to have multiple copies of the APK extracted in the system.
```

Comment:
Depending on how many times the app has been installed, Android keeps around the previous directories, but appends an integer suffix.

Neither shell nor the app's user have read access on `/data/app`, so it's not possible to see what directories exist. We could try iterating say for `indices <= 10`, and try accessing gdbserver in there, but it seems way more hacky than just copying gdbserver ourselves.